tests/Doctrine/Tests/DBAL/Query/Expression/CompositeExpressionTest.php

```diff
@@ -77,4 +77,12 @@ public function provideDataForConvertToString()
             ),
         );
     }
+
+    public function testAddPreventsAddingEmptyCompositeExpressionObjects()
+    {
+        $emptyExpr = new CompositeExpression(CompositeExpression::TYPE_AND, []);
+        $expr = new CompositeExpression(CompositeExpression::TYPE_AND, [$emptyExpr]);
+
+        $this->assertEquals(0, count($expr));
```

Comment:
Sorry, but I don't get your comment at all. I can try to explain the issue a bit more, but maybe there is any way to get in contact with you in any chat?

The problem which I ran into:
- reading the (old) code, it should be possible to pass an object of type CompositeExpression (object2) as a new part in another CompositeExpression object (object1)
- object2 should only be added it it contains parts on its own
- so creating object2 with empty parts and passing it to object1 shouldn't change parts of object1

Currently this is broken and an empty object2 is added to the parts of object1. IMHO this is not intended. I created the patch and a unit test which fails before and is fixed with the patch.